### ISSUE TITLE
Azurerm automation account - dsc configuraion JSON parse error

### ISSUE THREAD

**Issue by KISMIDEV** [hashicorp/terraform-provider-azurerm#15526]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform (and AzureRM Provider) Version

* Terraform Core version: v1.0.5
* AzureRM Provider version: v2.97.0

### Terraform Configuration Files

```hcl
resource "azurerm_automation_dsc_configuration" "basic" {
  for_each = {
    for configuration in flatten([
      for configuration_key, configuration_value in var.automation_account_basic : [
        for key, value in configuration_value.configuration : {
          configuration_name      = key
          content_embedded        = value.content_embedded
          automation_account_name = configuration_key
        }
      ] 
    ]) : "${configuration.configuration_name}" => configuration
  }

  name                    = each.value.configuration_name
  resource_group_name     = azurerm_resource_group.basic.name
  automation_account_name = azurerm_automation_account.basic[each.value.automation_account_name].name
  location                = azurerm_resource_group.basic.location
  content_embedded        = var.content_embedded
}

variable "content_embedded" {
    type = string
}
```

### Description / Feedback

Hi!
5 days ago, I already created an issue for Automation Accounts.
I received a reply that it was a duplicate of the issue number #10858 
I tested it again using the method listed there.
Test code is below.

`TF_VAR_content_embedded='Configuration default {}' terraform plan`
`TF_VAR_content_embedded='Configuration default {}' terraform apply`

I got the same result. So, I create issue again.

![image](https://user-images.githubusercontent.com/70998863/154922291-724033d8-8d0a-4cc0-9ca7-991b0646d78b.png)

To recap, I'm testing creating an `azurerm_automation_account` with other resources.
Below is a list of resources to be deployed together.
- virtual network
- subnet
- virtual network perring
- network security group
- key vault
- storage account

My automation account tf code works normally in the local enviroment and when only this code is loaded.
When combined with other code, the above error occurs.
This time, please, let me know how to solve it.
Below is my previous issue.
https://github.com/hashicorp/terraform-provider-azurerm/issues/15452

Thank you!

**Comment by neil-yechenwei** [comment]
@KISMIDEV , thanks for raising this issue. This is the base layer issue, which needs to be fixed in the base layer, as this isn't being deserialized correctly - https://github.com/hashicorp/terraform-provider-azurerm/issues/10858 is tracking this issue. So it's duplicate with it. The similar issue also could be found at https://github.com/hashicorp/terraform-provider-azurerm/issues/15404.

**Comment by mybayern1974** [comment]
@KISMIDEV , per neil-yechenwei's suggest, are you ok if we close this duplicate issue and you can subscribe #10858 ?

**Comment by tombuildsstuff** [comment]
Duplicate of #15404 which'll be fixed by #10858
